### PR TITLE
Eli Kalish Assignment5

### PR DESCRIPTION
**Pull request by elik723**

### CODE REVIEW

File: assignment5.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+1. How would you call the following function? Give an example of what it would look like written
+inside of the main() function and explain your code in detail.
+	int fxn(int a, float b, int c)
+	Assuming that the above function works, here's what I would code for the main function:
+	int main()
+	{
```
Comment (oldclesleycode) [inline assignment5.txt:6]: Perfect!

File: assignment5.txt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	also has a return type of int, so I created another variable of type int to store this
+	returned value. I then returned zero so the ANSI gods would be appeased.
+2. Explain the difference between recursion and iteration. When/why is one preferable over the other?
+Why is recursion worse in other cases?
+	Recursion is when a function calls upon itself, creating an infinite sequence unless there
+	is some base case specified in which the function no longer calls itself. Iteration is
+	basically loops, like a for loop or a while loop, etc. There are a couple of reasons
+	why one would be preferable over the other. In some cases, one is more efficient than
+	the other. Oftentimes, recursive functions look cleaner and more elegant than writing out
+	a nasty dirty for loop so one would use a recursive function then.
```
Comment (oldclesleycode) [inline assignment5.txt:23]: This is all right, but it's also important to note that recursive calls take up more memory. now that we're learning about memory management and data structures, you can further understand this by understanding that each time a function is called, a stack is initialized. So having a recursive function can be very memory ineffective.

File: assignment5.txt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+3. How does a compiler work? Explain step by step.
+	I don't know the specifics step by step, but what a compiler does is take the source
+	code that the user has written and converts it into machine instructions that the computer
```
Comment (oldclesleycode) [inline assignment5.txt:26]: right - so the process is the compiler breaking the text up by white space, converting this into a parse tree, optimizing said parse tree, and then translating that into machine instructions.

File: recurseFibonacci.c [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	}
+}
+
+int main()
+{
+	printf("How many numbers of the Fibonacci Sequence should be printed?\n");
+	int k;
+	scanf("%d", &k);
+	k -= 2;
+	printf("1 1 ");
```
Comment (lesley2958) [inline recurseFibonacci.c]: What about if the user only wants 0 or 1 of the fibonacci numbers?? Also, where are your comments?

File: loopFibonacci.c [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#include <stdio.h>
+
+int main()
+{
+	printf("How many numbers in the fibonacci series do you want to print?\n");
+	int k;
+	scanf("%d", &k);
+	int first = 1, second = 1, sum;
+	printf("1 1 ");
```
Comment (oldclesleycode) [inline loopFibonacci.c:9]: What about if the user only wants 0 or 1 of the fibonacci numbers?? Also, where are your comments?

File: recurseFibonacci.c (hunk 4 above)
Comment (oldclesleycode) [inline recurseFibonacci.c:22]: Again, you didn't think about edge cases. And again, comments!

**Pull request closed without merging**